### PR TITLE
[Merged by Bors] - chore(Probability/Kernel/CondCdf): cleanup

### PR DESCRIPTION
**Pull request by mo271**
- [x] depends on: #10298 
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Real/Archimedean.lean [hunk 1]
```diff
@@ -346,4 +347,24 @@ theorem iInf_Ioi_eq_iInf_rat_gt {f : ℝ → ℝ} (x : ℝ) (hf : BddBelow (f ''
       norm_cast
 #align infi_Ioi_eq_infi_rat_gt Real.iInf_Ioi_eq_iInf_rat_gt
 
+theorem notBddAbove_coe: ¬(BddAbove <| range (fun (x : ℚ) ↦ (x : ℝ))) := by
```
Comment (RemyDegenne) [inline Mathlib/Data/Real/Archimedean.lean:350]:
```suggestion
theorem not_bddAbove_coe : ¬ (BddAbove <| range (fun (x : ℚ) ↦ (x : ℝ))) := by
```

File: Mathlib/Data/Real/Archimedean.lean [hunk 2]
```diff
@@ -346,4 +347,24 @@ theorem iInf_Ioi_eq_iInf_rat_gt {f : ℝ → ℝ} (x : ℝ) (hf : BddBelow (f ''
...
+  dsimp only [BddAbove, upperBounds]
+  rw [Set.not_nonempty_iff_eq_empty]
+  ext
+  simpa using  exists_rat_gt _
```
Comment (RemyDegenne) [inline Mathlib/Data/Real/Archimedean.lean:354]:
```suggestion
  simpa using exists_rat_gt _
```

File: Mathlib/Data/Real/Archimedean.lean [hunk 3]
```diff
@@ -346,4 +347,24 @@ theorem iInf_Ioi_eq_iInf_rat_gt {f : ℝ → ℝ} (x : ℝ) (hf : BddBelow (f ''
...
+
+theorem notBddBelow_coe: ¬(BddBelow <| range (fun (x : ℚ) ↦ (x : ℝ))) := by
```
Comment (RemyDegenne) [inline Mathlib/Data/Real/Archimedean.lean:356]:
```suggestion
theorem not_bddBelow_coe : ¬ (BddBelow <| range (fun (x : ℚ) ↦ (x : ℝ))) := by
```

**Review by RemyDegenne — COMMENTED** [review]
Can you comment on the change of imports? Does that add many files?

File: Mathlib/Data/Real/Archimedean.lean [hunk 4]
```diff
@@ -346,4 +347,24 @@ theorem iInf_Ioi_eq_iInf_rat_gt {f : ℝ → ℝ} (x : ℝ) (hf : BddBelow (f ''
...
+  simpa using exists_rat_gt _
+
+theorem not_bddBelow_coe : ¬ (BddBelow <| range (fun (x : ℚ) ↦ (x : ℝ))) := by
+  dsimp only [BddBelow, lowerBounds]
+  rw [Set.not_nonempty_iff_eq_empty]
+  ext
+  simpa using exists_rat_lt _
+
+theorem iUnion_Iic_rat : ⋃ r : ℚ, Iic (r : ℝ) = univ := by
+  exact iUnion_Iic_of_not_bddAbove_range notBddAbove_coe
```
Comment (mo271) [inline Mathlib/Data/Real/Archimedean.lean:363]:
```suggestion
  exact iUnion_Iic_of_not_bddAbove_range not_BddAbove_coe
```

File: Mathlib/Data/Real/Archimedean.lean [hunk 5]
```diff
@@ -346,4 +347,24 @@ theorem iInf_Ioi_eq_iInf_rat_gt {f : ℝ → ℝ} (x : ℝ) (hf : BddBelow (f ''
...
+#align real.Union_Iic_rat Real.iUnion_Iic_rat
+
+theorem iInter_Iic_rat : ⋂ r : ℚ, Iic (r : ℝ) = ∅ := by
+  exact iInter_Iic_eq_empty_iff.mpr notBddBelow_coe
```
Comment (mo271) [inline Mathlib/Data/Real/Archimedean.lean:367]:
```suggestion
  exact iInter_Iic_eq_empty_iff.mpr not_BddBelow_coe
```

**Comment by mo271** [comment]
> Can you comment on the change of imports? Does that add many files?

I replaced import `Mathlib.Data.Set.Image` with `Mathlib.Data.Set.Intervals.Disjoint` and sorted the import alphabetically. Not sure how many more files that makes. Is there an easy way to list all imports of a file recursively?

File: Mathlib/Data/Real/Archimedean.lean [hunk 6]
```diff
@@ -346,4 +347,24 @@ theorem iInf_Ioi_eq_iInf_rat_gt {f : ℝ → ℝ} (x : ℝ) (hf : BddBelow (f ''
...
+  simpa using exists_rat_gt _
+
+theorem not_bddBelow_coe : ¬ (BddBelow <| range (fun (x : ℚ) ↦ (x : ℝ))) := by
+  dsimp only [BddBelow, lowerBounds]
+  rw [Set.not_nonempty_iff_eq_empty]
+  ext
+  simpa using exists_rat_lt _
+
+theorem iUnion_Iic_rat : ⋃ r : ℚ, Iic (r : ℝ) = univ := by
+  exact iUnion_Iic_of_not_bddAbove_range not_BddAbove_coe
```
Comment (mo271) [inline Mathlib/Data/Real/Archimedean.lean:363]:
```suggestion
  exact iUnion_Iic_of_not_bddAbove_range not_bddAbove_coe
```

File: Mathlib/Data/Real/Archimedean.lean [hunk 7]
```diff
@@ -346,4 +347,24 @@ theorem iInf_Ioi_eq_iInf_rat_gt {f : ℝ → ℝ} (x : ℝ) (hf : BddBelow (f ''
...
+  rw [Set.not_nonempty_iff_eq_empty]
+  ext
+  simpa using exists_rat_lt _
+
+theorem iUnion_Iic_rat : ⋃ r : ℚ, Iic (r : ℝ) = univ := by
+  exact iUnion_Iic_of_not_bddAbove_range not_bddAbove_coe
+#align real.Union_Iic_rat Real.iUnion_Iic_rat
+
+theorem iInter_Iic_rat : ⋂ r : ℚ, Iic (r : ℝ) = ∅ := by
+  exact iInter_Iic_eq_empty_iff.mpr not_BddBelow_coe
```
Comment (mo271) [inline Mathlib/Data/Real/Archimedean.lean:367]:
```suggestion
  exact iInter_Iic_eq_empty_iff.mpr not_bddBelow_coe
```

**Comment by RemyDegenne** [comment]
I asked around how to check imports: `lake exe graph --to X --from Y out.pdf` can give a view of the imports between files, one you have cache. In this case, it looks like you added almost nothing.

**Comment by RemyDegenne** [comment]
The command I listed gives only the files between the two files in the command. That's indeed not the same as the files additionally imported.
I don't understand the list you posted. Can you explain the format `"Mathlib.Tactic.GuardHypNums" -> "Mathlib.Tactic.Common";` ?

**Comment by mo271** [comment]
Yeah let me explain, I arrived at the list in the gist with the following commands:
```
lake exe graph --reduce  --to Mathlib.Data.Set.Image all_image.dot
lake exe graph --reduce  --to Mathlib.Data.Set.Intervals.Disjoint  all_disjoint.dot
sort all_disjoint.dot > all_disjoint.dot.sorted
sort all_image.dot > all_image.dot.sorted
comm -23 all_disjoint.dot.sorted all_image.dot.sorted
```
So this basically list all additional edges in the import graph

**Comment by mo271** [comment]
by the way, I think I will propose to add functionality to the ImportGraph package to just get a list of imports as a sorted list instead of the entire graph...

**Comment by RemyDegenne** [comment]
If I understand correctly, the interesting info in the list you posted is the list of files that appear in the list of edges. Those are the files imported by the new import and not the old one.

But what really interest us is the difference between the list of files imported by the union of all new imports, minus the list of files imported by the union of all old imports. It might be that Mathlib.Data.Set.Intervals.Disjoint imports many more files than Mathlib.Data.Set.Image but those files might be mostly imported by other import lines anyway.

**Comment by mo271** [comment]
That makes sense, calculating only the new edges by running
```
lake exe graph --reduce --to Mathlib.Data.Real.Archimedean all.dot
```
twice once with the new imports in place and once with only the old imports (building lake build `Mathlib.Data.Real.Archimedean` in between) gives only this diff 
```
  "Mathlib.Data.Set.Intervals.Disjoint" -> "Mathlib.Data.Real.Archimedean";
  "Mathlib.Data.Set.Lattice" -> "Mathlib.Data.Set.Intervals.Disjoint";
```
So the only extra file seems to be `Mathlib.Data.Set.Lattice`

File: Mathlib/Data/Real/Archimedean.lean [hunk 8]
```diff
@@ -3,10 +3,11 @@ Copyright (c) 2018 Mario Carneiro. All rights reserved.
...
 Authors: Mario Carneiro, Floris van Doorn
 -/
-import Mathlib.Data.Real.Basic
-import Mathlib.Data.Set.Image
-import Mathlib.Algebra.Order.Archimedean
 import Mathlib.Algebra.Bounds
+import Mathlib.Algebra.Order.Archimedean
+import Mathlib.Data.Real.Basic
+import Mathlib.Data.Set.Intervals.Disjoint
+
```
Comment (RemyDegenne) [inline Mathlib/Data/Real/Archimedean.lean:10]:
```suggestion
```

**Review by RemyDegenne — APPROVED** [review]
Great! Thanks for investigating the import issue! It would be very nice to get a bot to do that for us on PRs.
bors d+

**Pull request closed without merging**